agreement/actions.go

```diff
@@ -151,7 +151,7 @@ func (a networkAction) do(ctx context.Context, s *Service) {
 	case protocol.ProposalPayloadTag:
 		msg := a.CompoundMessage
 		payload := transmittedPayload{
-			unauthenticatedProposal: msg.Proposal,
+			unauthenticatedProposal: msg.Proposal.Compressed(),
```

Comment:
I think that you would want to have the digest in the vote include the "Compressed" proposal.